### ISSUE TITLE
`export default class` throws reference error when the class name is referenced inside the declaration

### ISSUE THREAD

**Issue by Arnavion** [Microsoft/TypeScript#2538]
This compiles and the output is correct.

```typescript
class One {
    foo(): One {
        return this;
    }
}

export default One;
```

```javascript
var One = (function () {
    function One() {
    }
    One.prototype.foo = function () {
        return this;
    };
    return One;
})();
exports.default = One;
```

---

Combining the class declaration and export doesn't compile:

```typescript
export default class One {
    foo(): One {
        return this;
    }
}
```

```
module1.ts(2,12): error TS2304: Cannot find name 'One'.
```

**Comment by csnover** [comment]
This is still broken for declaration files.

```ts
declare module 'foo' {
  export default class One {
      foo(): One;
  }
}
```

Repro: tsc foo.d.ts
Expected: No error
Actual: foo.d.ts(3,14): error TS2304: Cannot find name 'One'.
Version: fb44a23b758f212c722360fbc4790c539d55bf94

**Comment by mhegazy** [comment]
@csnover sorry for the hassle. can you give it another attempt.
